pilot/pkg/model/jwks_resolver.go

```diff
@@ -236,48 +240,33 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 		}
 		return e.pubKey, nil
 	}
-
-	var err error
-	var pubKey string
-	if jwksURI == "" {
-		// Fetch the jwks URI if it is not hardcoded on config.
-		jwksURI, err = r.resolveJwksURIUsingOpenID(issuer)
-	}
-	if err != nil {
-		log.Errorf("Failed to jwks URI from %q: %v", issuer, err)
-	} else {
-		var resp []byte
-		resp, err = r.getRemoteContentWithRetry(jwksURI, networkFetchRetryCountOnMainFlow)
-		if err != nil {
-			log.Errorf("Failed to fetch public key from %q: %v", jwksURI, err)
-		}
-		pubKey = string(resp)
-	}
-
 	r.keyEntries.Store(key, jwtPubKeyEntry{
```

Comment:
Why do you store an empty jwt pub key

Reply:
I am storing it for `jwksuri`, we have to store some entry in the cache so that it will be available in the background.

Reply:
So in this case, you can use the fake jwks

Reply:
there might be a case when `jwksuri` itself is not there, we are fetching is as well https://github.com/istio/istio/blob/master/pilot/pkg/model/jwks_resolver.go#L244
We are doing this as well in the background.